test/with_dummyserver/test_trailing_dot.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+import typing
+
+from dummyserver.server import DEFAULT_CA
+from dummyserver.testcase import HTTPDummyProxyTestCase
+from urllib3.poolmanager import proxy_from_url
+
+
+class TestHTTPProxyManagerAndTrailingDot(HTTPDummyProxyTestCase):
+    http_url_with_dot: typing.ClassVar[str]
+    https_url_with_dot: typing.ClassVar[str]
+
+    @classmethod
+    def setup_class(cls) -> None:
+        super().setup_class()
+        cls.http_url_with_dot = f"http://{cls.http_host}.:{int(cls.http_port)}"
+        cls.https_url_with_dot = f"https://{cls.https_host}.:{int(cls.https_port)}"
+        cls.proxy_url = f"http://{cls.proxy_host}:{int(cls.proxy_port)}"
+        cls.https_proxy_url = f"https://{cls.proxy_host}:{int(cls.https_proxy_port)}"
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        super().teardown_class()
+
+    def test_basic_proxy(self) -> None:
```

Comment:
When I run this test against v2.0.2, it fails with an exception: `urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='localhost.', port=64015): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'localhost.'. (_ssl.c:992)")))`